src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessResilienceTest.java

```diff
@@ -0,0 +1,128 @@
+import uk.gov.hmcts.reform.wacaseeventhandler.services.DeadLetterQueuePeekService;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.azure.messaging.servicebus.implementation.ManagementConstants.MESSAGE_ID;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@AutoConfigureMockMvc(addFilters = false)
+@ActiveProfiles("db")
+@TestPropertySource(properties = {"azure.servicebus.enableASB-DLQ=true",
+    "azure.servicebus.connection-string="
+    + "Endpoint=sb://REPLACE_ME/;SharedAccessKeyName=REPLACE_ME;SharedAccessKey=REPLACE_ME",
+    "azure.servicebus.topic-name=test",
+    "azure.servicebus.subscription-name=test",
+    "azure.servicebus.ccd-case-events-subscription-name=test",
+    "azure.servicebus.retry-duration=2"})
+public class MessageReadinessResilienceTest {
+
+    @MockBean
+    private LaunchDarklyFeatureFlagProvider launchDarklyFeatureFlagProvider;
+
+    @MockBean
+    private TelemetryClient telemetryClient;
+
+    @Mock
+    private TelemetryContext telemetryContext;
+
+    @Mock
+    private OperationContext operationContext;
+
+    @MockBean
+    private DeadLetterQueuePeekService deadLetterQueuePeekService;
+
+    @Autowired
+    MessageReadinessConsumer messageReadinessConsumer;
+
+    @MockBean
+    CaseEventMessageRepository caseEventMessageRepository;
+
+    AtomicInteger count = new AtomicInteger();
+    AtomicBoolean isDone = new AtomicBoolean(false);
+
+    @BeforeEach
+    void setup() {
+        when(launchDarklyFeatureFlagProvider.getBooleanValue(any(), any())).thenReturn(true).thenReturn(true);
+        lenient().when(telemetryClient.getContext()).thenReturn(telemetryContext);
+        lenient().when(telemetryContext.getOperation()).thenReturn(operationContext);
+    }
+
+    @Test
+    void should_handle_database_outage_and_log_issue_when_getting_all_messages_in_new_state() {
+        doThrow(new JDBCConnectionException("An error occurred when getting all message in new state", null))
+            .when(caseEventMessageRepository)
+            .getAllMessagesInNewState();
+
+        await().ignoreException(Exception.class)
+            .pollInterval(5, SECONDS)
+            .atMost(60, SECONDS)
+            .untilAsserted(() -> {
+                messageReadinessConsumer.run();
+                //The purpose of the test: we should see retry when an error occurred.
+                //So after 2 trying test will pass
+                if (count.get() > 1) {
+                    isDone.set(true);
+                }
+                count.getAndIncrement();
+                assertTrue(isDone.get());
```

Comment:
Can't you simplify this with assertTrue(count.get() > 1). Or using untilTrue isntead of untilAsserted

Reply:
done